### PR TITLE
Allow exec on one-off containers created with run

### LINKED ISSUE #6549: docker-compose exec work on "run" containers
## Example use case

`docker-compose.yml`

```
version: "3.2"
services:
  foo:
    image: debian:9
    command: sleep 10000
```

1. `docker-compose run -d foo`
2. `docker-compose exec foo bash`

### Current Result:

```
ERROR: No container found for foo_1
```

### Desired result

It would be nice if not only did `docker-compose exec` check for containers created by `up`, but also `run`. This is already done for commands like `down` and `ps`

**Example:** `docker-compose ps`

```
          Name              Command     State   Ports
-----------------------------------------------------
foo_foo_run_382b47cedaf4   sleep 10000  Up 
```

### Alternative solutions

If for some reason mixing the `up`ed and `run`ed containers together in `exec` is incompatible with use cases/maintaining backwards compatibility, then specifying a flag that either "Only execs using `run` containers" or combines the list of `up` and `exec` containers. Which ever you think is best if this is needed.

### Current work around

I currently have to have to use a mix of `bash`, `docker-compose ps`, or `docker ps` commands to find the container I want to exec into

### PR DESCRIPTION
**Pull request by collin5**
Resolves #6549

### CODE REVIEW

**Comment by ijc** [comment]
Thank you @collin5 for this PR.

Could you explain a bit more the reasoning behind the changes here. I see you are adding a new label to these one off containers, which suggests this is needed for some reason but I'm unclear what that is. If it is just to mirror the change in `get_container` to filter on that label for oneoff containers? In which case why is the label needed, is it not possible to filter for all the oneoffs for some reason?

Also, there is a `noqa` comment in there -- why?

**Comment by collin5** [comment]
Hi @ijc, sorry for the delay on this. I'm adding the label to one-off containers because [get_container](https://github.com/docker/compose/blob/26e1a2dd31b41ff3bc8418de879a8a839a199548/compose/service.py#L226) expects a container `index` with the default being set to `1`. We are currently adding this label for all the containers created with `docker-compose up` here https://github.com/docker/compose/blob/26e1a2dd31b41ff3bc8418de879a8a839a199548/compose/service.py#L891-L897 This was not the case for the one-off containers created with `run` which is part or the reasons we couldn't retrieve them here https://github.com/docker/compose/blob/26e1a2dd31b41ff3bc8418de879a8a839a199548/compose/service.py#L230

Also, `noqa` is for the flake8 to ignore the module level import not being at top of file. Putting this at top introduces circular import errors so I thought this was a better approach to avoid refactoring files not related to this issue.

**Comment by mafrosis** [comment]
@collin5 @ijc Any progress on this? It's bugged me for a long time :D

File: compose/service.py [hunk 1]
```diff
@@ -234,7 +234,12 @@ def get_container(self, number=1):
         """Return a :class:`compose.container.Container` for this service. The
         container must be active, and match `number`.
         """
-        for container in self.containers(labels=['{0}={1}'.format(LABEL_CONTAINER_NUMBER, number)]):
+        from compose.project import OneOffFilter  # noqa
```
Comment (kingbuzzman) [inline compose/service.py:237]: This feels like an avoidable hack, can we move `OneOffFilter` into a `enums.py` (or equivalent)? Then we can import it from both `service.py` and `project.py` circumventing this import.

**Comment by glours** [comment]
Thanks for taking the time to create this issue/pull request!

Unfortunately, Docker Compose V1 has reached end-of-life and we are not accepting any more changes (except for security issues). Please try and reproduce your issue with Compose V2 or rewrite your pull request to be based on the v2 branch and create a new issue or PR with the relevant Compose V2 information.

**Pull request closed without merging**